src/app/results/results.component.ts

```diff
@@ -213,6 +210,21 @@ export class ResultsComponent implements OnInit {
       this.presentPage = Math.abs(query['start'] / urldata.rows) + 1;
       let querydata = Object.assign({}, urldata);
       this.store.dispatch(new queryactions.QueryServerAction(querydata));
+        this.getJsonService.getJSON().subscribe(res => { this.newsResponse = [];
+          for (let i = 0; i < res.newsOrgs.length; i++) {
+          this.getNewsService.getSearchResults(querydata, res.newsOrgs[i].provider).subscribe(
+            response => {
+                if (typeof response.channels[0].items[0] !== 'undefined') {
+                this.newsResponse.push(response.channels[0].items[0]);
+                }
+                if (typeof response.channels[0].items[1] !== 'undefined') {
```

Comment:
Why using typeof if you are comparing it to check if its undefined or not. It will be string only.

Reply:
See above comment.